app/models/spotlight/resource.rb

```diff
@@ -42,24 +44,38 @@ def reindex_with_lock
       end
     end
 
+    def reindex_with_logging
+      time_start = Time.zone.now
+
+      count = yield
+
+      time_end = Time.zone.now
+
+      update(indexed_at: Time.current,
+             last_indexed_estimate: documents_to_index.size,
+             last_indexed_count: count,
+             last_index_elapsed_time: time_end - time_start)
+    end
+
     ##
     # Index the result of {#to_solr} into the index in batches of {#batch_size}
+    #
+    # @return [Integer] number of records indexed
     def reindex
```

Comment:
if we always call something called "reindex" then isn't it really "index" or "write_to_solr" or something?  If there's no diff between the first write and subsequent updates?